contracts/Delay.sol

```diff
@@ -62,108 +77,182 @@ contract Delay is AragonApp, IForwarder {
 
     /**
-    * @notice Resume a paused script execution with ID `_delayedScriptId`
-    * @param _delayedScriptId The ID of the script execution to resume
+    * @dev Allow script execution, IDisputable interface conformance
+    * @param _delayedScriptId The ID of the script execution to allow
     */
-    function resumeExecution(uint256 _delayedScriptId) external auth(RESUME_EXECUTION_ROLE) {
-        require(_isExecutionPaused(_delayedScriptId), ERROR_CAN_NOT_RESUME);
+    function _onDisputableActionAllowed(uint256 _delayedScriptId) internal {
         DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
-
         uint64 timePaused = getTimestamp64().sub(delayedScript.pausedAt);
-        delayedScript.executionTime = delayedScript.executionTime.add(timePaused);
+
+        delayedScript.executionFromTime = delayedScript.executionFromTime.add(timePaused);
         delayedScript.pausedAt = 0;
+        delayedScript.delayedScriptStatus = DelayedScriptStatus.Active;
 
-        emit ExecutionResumed(_delayedScriptId);
+        emit ExecutionResumed(_delayedScriptId, delayedScript.actionId);
     }
 
     /**
-    * @notice Cancel script execution with ID `_delayedScriptId`
-    * @param _delayedScriptId The ID of the script execution to cancel
+    * @dev Reject script execution, IDisputable interface conformance
+    * @param _delayedScriptId The ID of the script execution to reject
     */
-    function cancelExecution(uint256 _delayedScriptId) external scriptExists(_delayedScriptId) auth(CANCEL_EXECUTION_ROLE) {
-        delete delayedScripts[_delayedScriptId];
+    function _onDisputableActionRejected(uint256 _delayedScriptId) internal {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+
+        delayedScript.delayedScriptStatus = DelayedScriptStatus.Cancelled;
+
+        emit ExecutionCancelled(_delayedScriptId, delayedScript.actionId);
+    }
 
-        emit ExecutionCancelled(_delayedScriptId);
+    /**
+    * @dev Void script execution, IDisputable interface conformance
+    * @param _delayedScriptId The ID of the script execution to void
+    */
+    function _onDisputableActionVoided(uint256 _delayedScriptId) internal {
+        _onDisputableActionAllowed(_delayedScriptId);
     }
 
     /**
     * @notice Execute the script with ID `_delayedScriptId`
     * @param _delayedScriptId The ID of the script to execute
     */
-    function execute(uint256 _delayedScriptId) external {
-        require(canExecute(_delayedScriptId), ERROR_CAN_NOT_EXECUTE);
-
-        DelayedScript memory delayedScript = delayedScripts[_delayedScriptId];
-        delete delayedScripts[_delayedScriptId];
+    function execute(uint256 _delayedScriptId) external nonReentrant {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+        require(_canExecute(delayedScript), ERROR_CAN_NOT_EXECUTE);
 
         runScript(delayedScript.evmCallScript, new bytes(0), new address[](0));
```

Comment:
I think I would feel safer running this at the end.
Additionally, you may want to blacklist the agreement app.

Reply:
Okay, will do. However, can you outline what the risk is here taking into account `nonReentrant` modifier? Perhaps there's something I'm missing.